include/slang/syntax/SyntaxVisitor.h

```diff
@@ -109,6 +109,10 @@ class SyntaxRewriter : public SyntaxVisitor<TDerived> {
     ///
     /// @return if no changes are requested, returns the original syntax tree.
     /// Otherwise, the changes are applied and the newly rewritten syntax tree is returned.
+    ///
+    /// By default transformed tree holds shared_ptr to original one, preventing it from
+    /// being freed as long as its child is in scope. If you don't depend on this behaviour,
+    /// you can pass false as @a shareLifetimes, in order to save some memory.
```

Comment:
"shareLifetimes" isn't an argument here; I assume you changed that at some point to the member variable?

Reply:
Yeah, indeed. I guess I will remove this comment entirely since it duplicates docs of `setHoldParentPtr()` method